assets/shared/helpers/player/videopress-adapter.js

```diff
@@ -18,25 +18,42 @@ export const EMBED_PATTERN = /(videopress|video\.wordpress)\.com\/.+/i;
  */
 export const initializePlayer = ( element, w = window ) =>
 	new Promise( ( resolve ) => {
-		const onDurationChange = ( event ) => {
-			if (
-				event.source !== element.contentWindow ||
-				event.data.event !== 'videopress_durationchange' ||
-				! event.data.durationMs
-			) {
+		if ( element.dataset.duration ) {
+			resolve( element );
+		}
+
+		const onVideoPressMessage = ( event ) => {
+			if ( event.source !== element.contentWindow ) {
 				return;
 			}
 
-			// Set the duration to a dataset in order to have it available for later.
-			element.dataset.duration =
-				parseInt( event.data.durationMs, 10 ) / 1000;
+			const { data } = event;
 
-			w.removeEventListener( 'message', onDurationChange );
-			resolve( element );
+			if (
+				data.event === 'videopress_durationchange' &&
+				data.durationMs
+			) {
+				// Set the duration to a dataset in order to be available later,
+				// and consider the initialization done.
+				element.dataset.duration = data.durationMs / 1000;
+
+				// If current time didn't return yet, set it to `0`.
+				if ( ! element.dataset.currentTime ) {
+					element.dataset.currentTime = 0;
+				}
+
+				resolve( element );
+			} else if (
+				data.event === `videopress_timeupdate` &&
+				data.currentTimeMs
+			) {
+				// Set the current time to a dataset in order to be available later.
+				element.dataset.currentTime = data.currentTimeMs / 1000;
```

Comment:
It's also setting the `currentTime` to a dataset in the `iframe`, so we can get it without depending on the event (a case where the video is paused, and we need to get the current time, for example).